src/view/main_window.py

```diff
@@ -0,0 +1,131 @@
+# coding=utf-8
+"""main Window for the Editor"""
+
+from PySide6 import QtWidgets, QtGui
+
+from DMXModel import BoardConfiguration
+from Network import NetworkManager
+from Style import Style
+from model.broadcaster import Broadcaster
+from view.direct_mode.direct_scene_selector import DirectSceneSelector
+from view.filter_mode.NodeEditor import NodeEditorWidget
+from view.logging_mode.logging_widget import LoggingWidget
+from view.main_widget import MainWidget
+from view.patching_mode.patching_selector import PatchingSelector
+
+
+class MainWindow(QtWidgets.QMainWindow):
+    """Main window of the app. All widget are children of its central widget."""
+
+    def __init__(self, parent=None) -> None:
+        """Inits the MainWindow.
+
+        Args:
+            parent: Qt parent of the widget.
+        """
+        super().__init__(parent)
+        # first logging to don't miss logs
+        debug_console = LoggingWidget()
+        self._broadcaster = Broadcaster()
+        self.setWindowTitle("Project-Editor")
+
+        # model objects
+        self._fish_connector: NetworkManager = NetworkManager(self._broadcaster, self)
+        self._board_configuration: BoardConfiguration = BoardConfiguration()
+
+        # views
+        views: list[tuple[str, QtWidgets]] = [
+            ("Direct Mode", MainWidget(DirectSceneSelector(self._broadcaster, self), self)),
+            ("Filter Mode", MainWidget(NodeEditorWidget(self, self._board_configuration), self)),
```

Comment:
Filter Editor. The Show UI is still w.i.p.. Compare comment from other file. Just because it has a button to update the current running show does not mean that is a viable user interface to operate a show on.